src/tensorflow.cc

```diff
@@ -1280,31 +1273,37 @@ AutoCompleteHelper::FixBatchingSupport()
               } else {
                 config_io.MemberAsArray("dims", &config_dims);
               }
-              if (config_dims.ArraySize() != 0) {
-                // look up corresponding io info from model
-                for (const TRITONTF_IOList* itr = model_ios[ios_idx];
-                     itr != nullptr; itr = itr->next_) {
-                  TRITONTF_IO* io = itr->io_;
-                  if (config_name == io->name_) {
-                    bool should_batch =
-                        (io->shape_->rank_ == (config_dims.ArraySize() + 1));
-                    // inconsistent hint
-                    if (config_batch_hint &&
-                        (model_support_batching_ != should_batch)) {
-                      return TRITONSERVER_ErrorNew(
-                          TRITONSERVER_ERROR_INTERNAL,
-                          std::string(
-                              "unable to autofill for '" +
-                              model_state_->Name() +
-                              "', model tensor configurations are "
-                              "contradicting " +
-                              "each other in terms of whether batching is "
-                              "supported")
-                              .c_str());
+              for (const TRITONTF_IOList* itr = model_ios[ios_idx];
+                   itr != nullptr; itr = itr->next_) {
+                TRITONTF_IO* io = itr->io_;
+                if (config_name == io->name_) {
+                  bool model_io_explicit = io->shape_->rank_ > 0;
+                  bool user_config_is_defined = config_dims.ArraySize() > 0;
+
+                  if (model_io_explicit && user_config_is_defined) {
+                    if (config_dims.ArraySize() == io->shape_->rank_) {
+                      int64_t first_config_dim = 0;
+                      config_dims.IndexAsInt(0, &first_config_dim);
+                      if (first_config_dim != -1) {
+                        config_batch_hint = false;
+                      }
+                    } else if (
+                        config_dims.ArraySize() - 1 != io->shape_->rank_) {
+                      // Defer error to validation
+                      config_batch_hint = false;
                     }
-                    config_batch_hint = true;
-                    model_support_batching_ = should_batch;
-                  }
+                  } else if (!model_io_explicit && user_config_is_defined) {
+                    int64_t first_config_dim = 0;
+                    config_dims.IndexAsInt(0, &first_config_dim);
+                    if (first_config_dim != -1) {
```

Comment:
`user_config_dim` do not include the batch dimensions when batching is enabled. Hence, these cases will not accomplish anything worthwhile. 
If  rank of one of the input is 0, then this is how we must handle the case:

1. Assume the model is non-batching in general.
2. If any one of the input's rank is equal to `1 + config_dims.ArraySize()` , then the model is batching.

We can not do anything else for `!model_io_explicit`.

Reply:
I see my misunderstanding, for this case which you have highlighted I thought that for the case `(!model_io_explicit && user_config_is_defined)`, the user could have supplied the full shape _with_ the batch dimension included. After re-reading the documentation I see this is not the case. In the user provided model configuration, the inputs refer to the shape of the tensor in the inference request _not necessarily_ the shape of the savedmodel. Only in the case of a savedmodel which does not support batching do the inference request dimensions match the savedmodel dimensions.

With this model of batching in place, I agree with you that there is nothing to be done when the savedmodel is not explicit but the user has provided the dimensions for the IO. Assume the savedmodel doesn't batch since we don't know if the model config dimensions exactly match the savedmodel dimensions.

What is difficult about this is that it contradicts another decision we made which is: In the case where all inputs are ambiguous (i.e. !model_io_explicit) but the user model config describes the all inputs in their full dimension, [we decided to allow batching](https://nvidia.slack.com/archives/D036DKMM465/p1652230502337589). Maybe I had this wrong from the outset and we don't allow batching for ambiguous models but instead allow variable sized inputs for singly batched models?

Reply:
After an offline discussion, we confirmed @tanmayv25's comment is correct and that we should be taking a conservative approach, in this case, for batch hinting (we shouldn't batch). 

>What is difficult about this is that it contradicts another decision we made which is: In the case where all inputs are ambiguous (i.e. !model_io_explicit) but the user model config describes the all inputs in their full dimension, [we decided to allow batching](https://nvidia.slack.com/archives/D036DKMM465/p1652230502337589).

This previous decision was based on the optimistic approach and we decided to discard this decision.

I will update this routine to not allow batching in this case. Here are a few examples which this functionality will conform to:
- For a saved model with all inputs as rank 0 and the user has provided dimensions,
  - batch hint is false
- For a saved model that has one or more inputs with rank > 0 and the user provided dimensions has size rank-1
  - batch hint is true
- For a saved  model which has a mixture of rank=N  inputs and user provided dimensions size={N, N-1}
  - This is conflicting information which the user has given us regarding batching.

Reply:
Updated.